aesara/tensor/basic_opt.py

```diff
@@ -1645,6 +1649,10 @@ def dimshuffled_alloc(i):
             new_i.append(i)
     new_i[assert_op_idx] = assert_op_in
 
+    # If this assert is triggered, it means we are recreating an equivalent graph
+    # which would result in a cyclical merge optimization
+    assert not all(new is old for new, old in zip(new_i, node.inputs))
```

Comment:
Are we not sure that the changes above will necessarily prevent this from occurring?  If not, perhaps we need to put the logic of this statement somewhere above.

Reply:
I kept finding more corner cases... Hopefully there are no others by this point.

Reply:
Couldn't the be used as a `return` condition as well?

Reply:
It could... but still means we have some flawed logic above, where the "feasibility" of the rewrite is supposed to take place. What do you think is best? This test has a couple of other asserts in place, and I have no idea of how unconventional that is.